src/main/java/step3/RacingGame.java

```diff
@@ -0,0 +1,56 @@
+package step3;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class RacingGame {
+
+    private final RacingSpec racingSpec;
+    private final List<Car> joinCarList = new ArrayList<>();
+    private final List<List<Integer>> racingLog = new ArrayList<>();
+
+    public RacingGame(RacingSpec racingSpec) {
+        if (racingSpec == null) {
+            throw new NullPointerException(ErrorMessage.RACING_SPEC_IS_NULL);
+        }
+        this.racingSpec = racingSpec;
+        initRacingGame();
+    }
+
+    private void initRacingGame() {
+        for (int i = 0; i < racingSpec.getCarCount(); i++) {
+            joinCar();
+        }
+    }
+
+    public void start() {
+        for (int i = 0; i < racingSpec.getMoveCount(); i++) {
+            doMoveStep();
+        }
+    }
+
+    private void doMoveStep() {
+        joinCarList.stream().forEach(car -> car.move());
```

Comment:
```suggestion
        joinCarList.stream()
        .forEach(car -> car.move());
```
Stream API는 . 마다 줄을 바꿔주시면 가독성이 좋아진답니다 :)